AssemblerLib/LocalToGlobalIndexMap.h

```diff
@@ -31,32 +35,31 @@ namespace AssemblerLib
 class LocalToGlobalIndexMap
 {
 public:
+#ifdef USE_PETSC
+    typedef MathLib::RowColumnIndices<PetscInt> RowColumnIndices;
```

Comment:
``MeshLib::NodePartitionMesh`` is independent of PETSc, and it could be used to other node partitioned approach. However, members of RowColumnIndices are passed to PETSc matrix and vector, where the type of PetscInt (defined as int in current version) is needed. 

Of course, we can have a conversion safe assertion for unsigned long to int in the constructor of MeshComponentMap. But I think it is unnecessary, because in seldom case, the number of nodes can reach  the number of 2^32-1. Besides, we can use option  --with-64-bit-indices.